### PR TITLE
Fix minor bugs and add missing server datasource

### PR DESCRIPTION
**Pull request by nvthongswansea**
This is a part of PR https://github.com/terraform-providers/terraform-provider-gridscale/pull/48
- Fix bugs caused by type mismatch.
- Fix bugs caused by errors which are not handled during setting values.
- Add missing server datasource and its docs.

### CODE REVIEW

File: gridscale/datasource_gridscale_server_test.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccdataSourceGridscaleServer_basic(t *testing.T) {
+	name := fmt.Sprintf("object-%s", acctest.RandString(10))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGridscaleIpv4DestroyCheck,
```
Comment (itakouna) [inline gridscale/datasource_gridscale_server_test.go]: `testAccCheckGridscaleIpv4DestroyCheck`  should be `testAccCheckResourceGridscaleServerDestroyCheck`.

Comment (itakouna) [reply gridscale/datasource_gridscale_server_test.go]: Thanks for splitting https://github.com/terraform-providers/terraform-provider-gridscale/pull/48

Comment (itakouna) [reply gridscale/datasource_gridscale_server_test.go]: Plz, would you fix this for `TestAccdataSourceGridscaleFirewall_basic` https://github.com/terraform-providers/terraform-provider-gridscale/blob/fa59c99170c7a502363eb74884edfe04eb447606/gridscale/datasource_gridscale_firewall_test.go#L17

Comment (nvthongswansea) [reply gridscale/datasource_gridscale_server_test.go]: @itakouna Done!

**Review by itakouna — APPROVED** [review]

**Review by bkircher — APPROVED** [review]
Looks good to me

**Pull request merged**